File: benchmarks/pandas/bench_index_min_max.py
"""Benchmark: index_min_max — Index.min and Index.max on 100k-element Index"""
import json, time
import pandas as pd
import numpy as np

SIZE = 100_000
WARMUP = 3
ITERATIONS = 20

labels = np.arange(SIZE)
idx = pd.Index(labels)

for _ in range(WARMUP):
    idx.min()
    idx.max()

start = time.perf_counter()
for _ in range(ITERATIONS):
    idx.min()
    idx.max()
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "index_min_max",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))

File: benchmarks/tsb/bench_index_min_max.ts
/**
 * Benchmark: index_min_max — Index.min and Index.max on 100k-element Index
 */
import { Index } from "../../src/index.js";

const SIZE = 100_000;
const WARMUP = 3;
const ITERATIONS = 20;

const labels = Array.from({ length: SIZE }, (_, i) => i);
const idx = new Index(labels);

for (let i = 0; i < WARMUP; i++) {
  idx.min();
  idx.max();
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  idx.min();
  idx.max();
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "index_min_max",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
